src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -829,11 +829,14 @@ self =>
         s"Incompatible args to makeBinop: !isExpr but targs=$targs")
 
       def mkSelection(t: Tree) = {
-        val pos = opPos union t.pos
+        val pos = (opPos | t.pos) makeTransparentIf !treeInfo.isLeftAssoc(op)
```

Comment:
perhaps `val rassoc = !treeInfo.isLeftAssoc(op)`, and reuse below.

Reply:
Done.